app/views.py

```diff
@@ -0,0 +1,149 @@
+import random
+from datetime import timedelta
+
+from django.contrib.auth.decorators import login_required
+from django.db.models.functions import Length
+from django.http import HttpRequest, HttpResponse, HttpResponseServerError
+from django.shortcuts import get_object_or_404, redirect, render, reverse
+from django.utils import timezone
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+
+from api.models import Submission, Transcription
+from app.permissions import require_coc, require_reddit_auth
+from utils.mixins import CSRFExemptMixin
+from website.helpers import get_additional_context
+
+
+@login_required
+@csrf_exempt
+def accept_coc(request: HttpRequest) -> HttpResponse:
+    """Show the Code of Conduct and ask the volunteer to accept it."""
+    if request.method == "POST":
+        request.user.accepted_coc = True
+        request.user.save()
+        return redirect(reverse("choose_transcription") + "?show_tutorial=1")
+    else:
+        return render(request, "app/accept_coc.html", get_additional_context())
+
+
+@login_required
+@require_coc
+@require_reddit_auth
+def choose_transcription(request: HttpRequest) -> HttpResponse:
+    """Provide a user with transcriptions to choose from."""
+    # time_delay = timezone.now() - timedelta(hours=settings.ARCHIVIST_DELAY_TIME)
+    time_delay = timezone.now() - timedelta(hours=60)
+    options = Submission.objects.annotate(original_id_len=Length("original_id")).filter(
+        original_id_len__lt=10,
+        completed_by=None,
+        claimed_by=None,
+        create_time__gte=time_delay,
+    )
+
+    if options.count() > 3:
+        # we have more to choose from, so let's grab 3
+        temp = []
+        while len(temp) < 3:
+            submission = random.choice(options)
+            # todo: we want to eventually handle other types of content here, but
+            #  we simply aren't ready for it yet.
+            if submission not in temp and submission.is_image:
+                temp.append(submission)
+        options = temp
+
+    for submission in options:
+        # todo: add check for source. This will need to happen after we convert
+        #  sources to the individual subreddits
+        if not submission.title:
+            # we have a lot of submissions that don't have a title, but we
+            # need it for this next page. Just grab the title and save it.
+            post = request.user.reddit.submission(url=submission.url)
+            submission.title = post.title
+            submission.nsfw = post.over_18
+            submission.save(skip_extras=True)
+
+    context = get_additional_context({"options": options, "fullwidth_view": True})
+
+    # messages.success(request, "You ranked up to Jade! Congrats!!")
+    # context.update({"show_confetti": True})
```

Comment:
We should probably implement this properly at some point, or replace it with a TODO comment for now (personally I'm all for keeping the confetti :D)

Reply:
it's implemented properly now :)